lib/runtime-core/src/typed_func.rs

```diff
@@ -819,8 +818,14 @@ mod tests {
                     vec![$($x),*].iter().sum()
                 }
 
-                let _func = Func::new(with_vmctx);
-                let _func = Func::new(without_vmctx);
+                let _ = Func::new(with_vmctx);
+                let _ = Func::new(without_vmctx);
+                let _ = Func::new(|_: &mut vm::Ctx, $($x: i32),*| -> i32 {
+                    vec![$($x),*].iter().sum()
+                });
+                let _ = Func::new(|$($x: i32),*| -> i32 {
+                    vec![$($x),*].iter().sum()
+                });
```

Comment:
Do we call these and assert the results in some tests somewhere?

Reply:
Assertions here are the type-system. If it compiles, it means `Func` supports the described functions.